### PR TITLE
openharmony: add instructions for servoshell

### PR DESCRIPTION
**Pull request by mukilan**
Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

### CODE REVIEW

File: src/hacking/building-for-openharmony.md [hunk 1]
```diff
@@ -105,15 +110,84 @@ Depending on your Linux distributions you may want to use a different group.
...
+        "storeFile": "/path/to/java-keystore-file.p12"
+    }
+}
+```
+
+Here <encrypted password> is a hexadecimal string representation of the plaintext password after being encrypted.
+The key and salt information used to encrypt these passwords are generated by DevEco Studio IDE and are stored on-disk alongside
+the certificate files and keystore, usually under `<USER HOME>/.ohos/config/openharmony`.
+
+You can use the IDE to generate information needed for password encryption, the required application and
```
Comment (jschwe) [inline src/hacking/building-for-openharmony.md:148]: Probably we should mention that this is only for developing / testing purposes. To ship an app, one would need to get the signature configuration from the app store provider.

File: src/hacking/building-for-openharmony.md [hunk 2]
```diff
@@ -105,15 +110,84 @@ Depending on your Linux distributions you may want to use a different group.
...
+valid `.json` file with the same structure as the `signingConfigs` property given above, but with `certPath`, `storeFile` and
+`profile` given as *paths relative to the json file*, instead of absolute paths.
+
+
+## Building Servo shell
+
+Before building servo, ensure the following environment variables are set in the current shell:
+
+```commandline
+    # Note: The openharmony sdk is under ${DEVECO_SDK_HOME}/HarmonyOS-NEXT-${HOS_VERSION}/openharmony
```
Comment (jschwe) [inline src/hacking/building-for-openharmony.md:165]: I think in Developer Beta 7 / Beta 1 this path was changed. I'd need to check next week, but I think it's now something like `${DEVECO_SDK_HOME}/default/openharmony`.

File: src/hacking/building-for-openharmony.md (hunk 1 above)
Comment (mukilan) [reply src/hacking/building-for-openharmony.md:148]: done in [4090da6](https://github.com/servo/book/pull/24/commits/4090da6a8f2698c7c1b68402252580a44c5187bd)

File: src/hacking/building-for-openharmony.md [hunk 3]
```diff
@@ -105,15 +110,84 @@ Depending on your Linux distributions you may want to use a different group.
...
+```
+
+The following command can then be used to compile the servo shell application for a 64-bit ARM device:
+
+```commandline
+./mach build --ohos --release
+```
+
+In `mach build`, `mach install` and `mach package` commands, `--ohos` is an alias for `--target aarch64-unknown-linux-ohos`.
+To build for the emulator, use `--target x86_64-unknown-linux-ohos`
```
[the remark this answers is not in the document]
Comment (mukilan) [reply src/hacking/building-for-openharmony.md]: Good catch! I've made the text more explicit that this command is for an emulator running on x86-64 hosts.

File: src/hacking/building-for-openharmony.md (hunk 2 above)
Comment (mukilan) [reply src/hacking/building-for-openharmony.md]: I've update the text to be `${DEVECO_SDK_HOME}/default/openharmony`. Is that different from one where commandline tools is used?

File: src/hacking/building-for-openharmony.md [hunk 4]
```diff
@@ -105,15 +110,84 @@ Depending on your Linux distributions you may want to use a different group.
...
+When using the `hvigor` tool, this can be accomplished by setting a static `signingConfigs` object in the `build-profile.json5` file or
+by dynamically creating the `signingConfigs` array on the appication context object using the `hvigorfile.ts` script.
+
+The `signingConfigs` property is an array of objects with the following structure:
+
+```json
+
+{
+    "name": "default",
+    "type": "OpenHarmony",
```
[the remark this answers is not in the document]
Comment (mukilan) [reply src/hacking/building-for-openharmony.md]: I've changed this to `"type": "<OpenHarmony or HarmonyOS>"`

File: src/hacking/building-for-openharmony.md (hunk 2 above)
Comment (jschwe) [reply src/hacking/building-for-openharmony.md]: `DEVECO_SDK_HOME` points to `/path/to/DevEcoStudio/sdk`, so that seems to match.

**Comment by delan** [comment]
Looks good so far, just remember to put one sentence on each line :)

**Comment by mukilan** [comment]
> Looks good so far, just remember to put one sentence on each line :)

Thanks! Addressed in [bbede1b](https://github.com/servo/book/pull/24/commits/bbede1b87571decf4cd4a4e3b990ff2e222ab35d).

**Comment by delan** [comment]
I’ll review this next week, after the monthly update :)

File: src/hacking/building-for-openharmony.md [hunk 5]
```diff
@@ -9,7 +9,7 @@ Support for OpenHarmony is currently in-progress and these instructions might ch
 Building for OpenHarmony requires the following:
 
 1. The OpenHarmony SDK. This is sufficient to compile servo as a shared library for OpenHarmony.
-2. The `hvigorw` build tool to compile apps into an app bundle and sign it.
+2. The `hvigor` build tool to compile apps into an app bundle and sign it.
```
Comment (mrobinson) [inline src/hacking/building-for-openharmony.md:12]:
```suggestion
2. The `hvigor` build tool to compile an application into an app bundle and sign iy.
```

File: src/hacking/building-for-openharmony.md [hunk 6]
```diff
@@ -63,37 +76,29 @@ care to install the hvigor version matching the requirements of your project.
 
 `hvigor` (not the wrapper `hvigorw`) is also available via `npm`.
 1. Install the same nodejs version as the commandline-tools ship.
-   For HarmonyOS NEXT Node 18 is shipped.
+   For HarmonyOS NEXT Node 18 is shipped. Ensure node binary is in PATH.
```
Comment (mrobinson) [inline src/hacking/building-for-openharmony.md:79]:
```suggestion
   For HarmonyOS NEXT Node 18 is shipped. Ensure that the `node` binary is in PATH.
```

File: src/hacking/building-for-openharmony.md [hunk 7]
```diff
@@ -63,37 +76,29 @@ care to install the hvigor version matching the requirements of your project.
...
+   /path/to/node_modules/@ohos/hvigor/bin/hvigor.js assembleHap
    ```
 
 ### Configuring hdc on Linux
 
 `hdc` is the equivalent to `adb` for OpenHarmony devices.
 You can find it in the `toolchains` directory of your SDK.
 For convenience purposes, you might want to add `toolchains` to your `PATH`.
-Among others, `hdc` can be used to open a shell or send/receive files from a device
+Among others,`hdc` can be used to open a shell or send/receive files from a device.
```
Comment (mrobinson) [inline src/hacking/building-for-openharmony.md:101]:
```suggestion
Among others,`hdc` can be used to open a shell or transfer files between a device and the host system.
```

**Review by mrobinson — APPROVED** [review]

File: src/hacking/building-for-openharmony.md [hunk 8]
```diff
@@ -105,15 +110,79 @@ Depending on your Linux distributions you may want to use a different group.
...
+        "keyAlias": "debugKey",
+        "keyPassword": "<encrypted password>",
+        "profile": "/path/to/signed-profile-certificate.p7b",
+        "signAlg": "SHA256withECDSA",
+        "storeFile": "/path/to/java-keystore-file.p12"
+    }
+}
+```
+
+Here <encrypted password> is a hexadecimal string representation of the plaintext password after being encrypted.
```
Comment (delan) [inline src/hacking/building-for-openharmony.md:142]:
```suggestion
Here `<encrypted password>` is a hexadecimal string representation of the plaintext password after being encrypted.
```

File: src/hacking/building-for-openharmony.md [hunk 9]
```diff
@@ -63,37 +76,29 @@ care to install the hvigor version matching the requirements of your project.
...
+   /path/to/node_modules/@ohos/hvigor/bin/hvigor.js assembleHap
    ```
 
 ### Configuring hdc on Linux
 
 `hdc` is the equivalent to `adb` for OpenHarmony devices.
 You can find it in the `toolchains` directory of your SDK.
 For convenience purposes, you might want to add `toolchains` to your `PATH`.
-Among others, `hdc` can be used to open a shell or send/receive files from a device
+Among others,`hdc` can be used to open a shell or transfer files between a device and the host system.
```
Comment (delan) [inline src/hacking/building-for-openharmony.md:101]:
```suggestion
Among others, `hdc` can be used to open a shell or transfer files between a device and the host system.
```

**Review by delan — APPROVED** [review]
Looks good with a couple of minor edits.

**Pull request merged**